src/idb.filesystem.js

```diff
@@ -37,13 +37,63 @@ if (exports.requestFileSystem || exports.webkitRequestFileSystem) {
 // Bomb out if no indexedDB available
 var indexedDB = exports.indexedDB || exports.mozIndexedDB ||
                 exports.msIndexedDB;
-if (!indexedDB)
-{
+if (!indexedDB) {
   return;
 }
 
-exports.TEMPORARY = 0;
-exports.PERSISTENT = 1;
+// Check to see if IndexedDB support blobs
+var support = new function() {
+  var dbName = "blob-support";
+  indexedDB.deleteDatabase(dbName).onsuccess = function() {
+    var request = indexedDB.open(dbName, 1.0);
+    request.onerror = function() {
+      support.blob = false;
+    };
+    request.onsuccess = function() {
+      var db = request.result;
+      try {
+        var blob = new Blob(["test"], {type:"text/plain"});
+        var transaction = db.transaction("store", "readwrite");
+        transaction.objectStore("store").put(blob, "key");
+        support.blob = true;
+      } catch (err) {
+        support.blob = false;
+      } finally {
+        db.close();
+        indexedDB.deleteDatabase(dbName);
+      }
+    };
+    request.onupgradeneeded = function() {
+      request.result.createObjectStore("store");
+    };
+  };
+};
+
+var Base64ToBlob = function(uri) {
+  uri = uri.split(',');
```

Comment:
can we use this impl: https://github.com/ebidel/filer.js/blob/master/src/filer.js#L137. It's well tested :)